src/ECER.Clients.RegistryPortal/ecer.clients.registryportal.client/src/store/loading.ts

```diff
@@ -0,0 +1,34 @@
+// Can be used in various components to check whether an API call is waiting for a response for loading spinners
+// Example in Application.vue we do this in a button :loading="loadingStore.isLoading('application_post')"
+// Loading states are set by our apiResultHandler.ts where we provide a key based on the API operations we have ex. application_post
+
+import { defineStore } from "pinia";
+
+import type { OperationMethods } from "@/types/openapi";
+
+export type LoadingOperation = keyof OperationMethods;
+
+export type LoadingState = {
+  [key in LoadingOperation]: boolean;
+};
+
+export interface LoadingStateConfig {
+  loadingStates: LoadingState;
+}
+
+export const useLoadingStore = defineStore("loading", {
+  state: (): LoadingStateConfig => ({
+    loadingStates: {} as LoadingState,
+  }),
+
+  getters: {
+    isLoading() {
+      return (key: LoadingOperation) => this.loadingStates[key];
+    },
+  },
+  actions: {
+    async setLoading(key: LoadingOperation, loading: boolean) {
+      this.$patch({ loadingStates: { [key]: loading } });
```

Comment:
patch is interesting, it allows us to just set the key and it'll retain all the other keys that were in there previously. 

it's similar to what we did in wizard 
this.$state.wizardData = {...this.$state.wizardData, newInfo: info};

Reply:
yeah it maintains all the other keys. I tested this by adding another property when we hit submit "draftapplication_put"

and I see that the key stays there 

![image](https://github.com/bcgov/ECC-ECER/assets/74216496/460d3cc6-4a3d-48b9-845e-3b352af9c5ca)